.github/workflows/actions.yml

```diff
@@ -54,6 +54,51 @@ jobs:
       run: |
         pip uninstall -y tensorflow-text tensorflow
         cd integration_tests && pytest . -k "NoTensorflow"
+  run_tests_with_keras_3_1_0:
```

Comment:
I worry this might be both too many jobs and a redundant with the block above. We actually update these configurations a fair amount.

Maybe we just do this for one backend? Torch cause its fastest? And I think we can further leverage a testing matrix to avoid duplicating this whole action block.

https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#example-adding-configurations

Something like this, probably needs tweaks.

```
    strategy:
      fail-fast: false
      matrix:
        backend: [tensorflow, jax, torch]
        version: [latest]
        include:
          - backend: torch
             version: 3.1
... later but before testing
    - name: Pin Keras Version
      if: ${{ matrix.version != '3.1'}}
      run: |
        pip uninstall -y keras
        pip install keras==3.1.0 --progress-bar off
```

Reply:
Sure! Updated.